docs/aro-hcp-debug-cheatsheet.md

```diff
@@ -0,0 +1,245 @@
+        managed_cluster_cr["ManagedCluster CR"]
+    end
+ end
+ rp_backend --> cs
+ arm --> istio_ingress --> rp_frontend --> cs -- manifests --> maestro_server -- ManifestWork --> maestro_agent -- ManifestWork --> acm --> hostedcluster_cr <--> hypershift
+ rp_frontend -- auth --> mise
+ istio_ingress --> mise
+ hypershift --> hypershift_cp_ns
+ acm --> managed_cluster_cr
+ hostedcluster_cr -- status updats --> managed_cluster_cr
+ hostedcluster_cr -- reconciled into --> cp_pods
+ maestro_agent -- status --> maestro_server
+
+ click rp_frontend "#rp-frontend" "Go to RP Frontend Section"
+ click rp_backend "#rp-backend" "Go to RP Backend Section"
+ click cs "#clusters-service" "Go to Clusters Service Section"
+ click maestro_agent "#maestro-agent" "Go to Maestro Agent Section"
+ click maestro_server "#maestro-server" "Go to Maestro Server Section"
+ click acm "#acm" "Go to ACM Section"
+ click hypershift "#hypershift" "Go to Hypershift Section"
+
+```
+
+## RP Frontend
+
+* The RP frontend is the primary regional API entry point for the ARO HCP service, as it receives requests from ARM
+* It delegates authentication to [MISE](../frontend/deploy/charts/mise) through an Istio [AuthorizationPolicy](../frontend/deploy/templates/ext-authz.authorizationpolicy.yaml)
+* It keeps track of cluster state in CosmosDB
+* It delegates the HCP creation to Clusters Service
+* Communication with Clusters Service is done via the Istio with mTLS and policies in place
+
+### Port Forward to the RP Frontend
+
+In order to access the RP frontend without going through ARM, you can port forward to the service. This is useful for debugging and testing purposes, especially in non-production environments where there is no ARM route to the service.
+Keep in mind that the port-forwarded service does not require authentication and does not enforce authorization!
+
+```sh
+kubectl port-forward -n aro-hcp svc/aro-hcp-frontend 8443:8443
+```
+
+### Query the State of an HCP
+
+You can query the state of a hosted control plane by sending a GET request to the port-forwarded service. The URL should include the resource ID of the hosted control plane you want to query, so something matching this:
+
+```sh
+RESOURCE_ID="/subscriptions/.../resourceGroups/.../providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters/..."
+curl -s localhost:8443${RESOURCE_ID}$?api-version=2024-06-10-preview | jq
+```
+
+Noteworthy fields:
+
+* `.id` - the resource ID of the hosted control plane
+* `.properties.provisioningState` - the provisioning state of the hosted control plane
+* `.properties.api.url` - the HCP KAS URL
+* `.properties.dns.baseDomain` - the base DNS zone for the HCP (exists in our infrastructure)
+
+## RP Backend
+
+* Updates the state of the hosted control plane in CosmosDB to support async operations
+
+## Clusters Service
+
+* orchestrates HCP creation by
+  * provisioning Azure cloud resources within the customer tenant and ARO HCP service tenants
```

Comment:
Which? Doesn't HyperShift do all of the Azure infra for each hosted control plane?

Reply:
i removed these kind of details from this document, but salved the bits for another document with exactly the purpose to give a component overview.